File: frontend/app/src/main/java/com/h1oo7/musicapp/model/Playlist.java
// model/Playlist.java
package com.h1oo7.musicapp.model;

import com.google.gson.annotations.SerializedName;
import java.util.ArrayList;
import java.util.List;

public class Playlist {
    @SerializedName("_id")
    private String id;
    private String name;
    private String user;
    private List<Song> songs;
    private boolean isPublic;
    private String createdAt;

    public String getId() { return id; }
    public String getName() { return name; }
    public List<Song> getSongs() { return songs != null ? songs : new ArrayList<>(); }
    public int getSongCount() { return getSongs().size(); }
}

File: forntend/app/src/main/java/com/example/musique/data/model/Playlist.kt
package com.example.musique.data.model

data class Playlist(
    val _id: String,
    val name: String,
    val description: String? = null,
    val userId: String,
    val songs: List<Song> = emptyList(),
    val isPublic: Boolean = true,
    val createdAt: String? = null,
    val updatedAt: String? = null
)
